onyo/lib/onyo.py

```diff
@@ -1066,9 +1100,13 @@ def set(self, paths: Iterable[Union[Path, str]], values: dict, dryrun: bool,
         if name_values:
             try:
                 self._update_names(assets_to_set, name_values)
-                self.clear_caches()
             except ValueError as e:
-                self.clear_caches()
+                self.clear_caches(assets=True,  # `onyo set` can rename assets
```

Comment:
Why are the caches cleared before doing the `restore()` (which clears the caches)?